app/server/appsmith-server/src/main/java/com/appsmith/server/exceptions/GlobalExceptionHandler.java

```diff
@@ -187,7 +222,22 @@ public Mono<ResponseDTO<ErrorDTO>> catchException(Exception e, ServerWebExchange
         AppsmithError appsmithError = AppsmithError.INTERNAL_SERVER_ERROR;
         exchange.getResponse().setStatusCode(HttpStatus.resolve(appsmithError.getHttpErrorCode()));
         doLog(e);
-        return Mono.just(new ResponseDTO<>(appsmithError.getHttpErrorCode(), new ErrorDTO(appsmithError.getAppErrorCode(),
+        String urlPath = exchange.getRequest().getPath().toString();
+        Mono<ResponseDTO<ErrorDTO>> response = Mono.just(new ResponseDTO<>(appsmithError.getHttpErrorCode(), new ErrorDTO(appsmithError.getAppErrorCode(),
                 appsmithError.getMessage())));
+
+        return getResponseDTOMono(urlPath, response);
+    }
+
+    private Mono<ResponseDTO<ErrorDTO>> getResponseDTOMono(String urlPath, Mono<ResponseDTO<ErrorDTO>> response) {
```

Comment:
As a practice, we should not pass `Mono` variables into functions. The signature of this function would be a lot cleaner if we did the following:

```suggestion
    private Mono<ResponseDTO<ErrorDTO>> getResponseDTOMono(String urlPath, ResponseDTO<ErrorDTO> response) {
```

Reply:
Yes makes sense. But I do have seen similar functions in the code base.